api/api-common/src/main/kotlin/io/qalipsis/api/steps/RateMeterStepSpecification.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.api.steps
+
+import io.micronaut.core.annotation.Introspected
+import io.qalipsis.api.context.StepContext
+import io.qalipsis.api.meters.Rate
+import io.qalipsis.api.meters.RateFailureConditionSpec
+import io.qalipsis.api.meters.RateFailureConditionSpecImpl
+import io.qalipsis.api.meters.specification.ComparableValueFailureSpecification
+
+/**
+ * Specification for a [io.qalipsis.core.factory.steps.meter.RateMeterStep].
+ *
+ * @author Francisca Eze
+ */
+@Introspected
+data class RateMeterStepSpecification<INPUT>(
+    val meterName: String,
+    val block: (stepContext: StepContext<INPUT, INPUT>, input: INPUT) -> TrackedThresholdRatio,
+) : AbstractStepSpecification<INPUT, INPUT, RateMeterStepSpecification<INPUT>>() {
+
+    var checks: MutableList<ComparableValueFailureSpecification<Rate, Double>> = mutableListOf()
+
+    fun shouldFailWhen(block: RateFailureConditionSpec.() -> INPUT): StepSpecification<*, INPUT, *> {
+        val rateFailureConditionSpec = RateFailureConditionSpecImpl()
+        rateFailureConditionSpec.block()
+        checks = rateFailureConditionSpec.checks
+        return this
+    }
+
+}
+
+/**
+ * Executes the block on the context to extract a value to be recorded in the meter.
+ *
+ * @param name the name of the meter.
+ * @param block the rule to convert the input and the context into the output.
+ *
+ * @author Francisca Eze
+ */
+fun <INPUT> StepSpecification<*, INPUT, *>.rate(
+    name: String,
+    block: (stepContext: StepContext<INPUT, INPUT>, input: INPUT) -> TrackedThresholdRatio
+): RateMeterStepSpecification<INPUT> {
+    val step = RateMeterStepSpecification(name, block)
+    this.add(step)
+    return step
+}
+
+data class TrackedThresholdRatio(val benchmark: Double, val total: Double)
```

Comment:
Please document what benchmark and total are. This will not be that obvious for our users. Do not hesitate to provide an example.